webapp/views/App/views/Data/Charts/state/hooks/useGetDimensionsFromArena/index.js

```diff
@@ -0,0 +1,34 @@
+import React, { useCallback, useEffect, useState } from 'react'
+
+import { Query } from '@common/model/query'
+import { useSurvey, useSurveyPreferredLang } from '@webapp/store/survey'
+
+import getArenaDimensions from './utils/getSelfAttributesAndAncestors'
+
+const useGetDimensionsFromArena = (nodeDefLabelType) => {
+  const survey = useSurvey()
+  const language = useSurveyPreferredLang()
+  const [entityDefUuid, setEntityDefUuid] = useState(null)
+  const [query, setQuery] = useState([])
+  const [dimensions, setDimensions] = useState([])
+
+  const generateDimensions = useCallback(() => {
+    const dimensions = getArenaDimensions({ nodeDefUuid: entityDefUuid, survey, language, nodeDefLabelType })
+    if (dimensions) {
+      setDimensions(dimensions)
+    }
```

Comment:
be careful with the "renaming" of this dimensions variable also declared on the line 13, 
use _dimensions or arenaDimensions as name of this variable,
also the if is not needed